App.js

```diff
@@ -157,212 +153,279 @@ const routeConfigs = {
     }
 
-    componentWillMount() {
+    this.sendAssignNotifs = this.sendAssignNotifs.bind(this)
+    this.sendUnassignNotifs = this.sendUnassignNotifs.bind(this)
+  }
+
+  componentWillMount() {
+
+    Settings.addListener('server:remove', this.disconnect.bind(this))
+
+    AppUser.load()
+      .then(user => {
+        Settings
+          .loadServer()
+          .then(baseURL => this.initializeRouter(baseURL, user))
+      })
+  }
+
+  createWebsocket(client) {
+    return new WebSocketClient(client, '/dispatch', {
+      onConnect: () => { events.emit('websocket:connect')},
+      onDisconnect: () => { events.emit('websocket:disconnect')},
+      onReconnect: () => { events.emit('websocket:reconnect')},
+      onMessage: this.onWebSocketMessage.bind(this),
+    })
+  }
+
+  onWebSocketMessage(event) {
+    // handle notification
+    let data = JSON.parse(event.data)
+
+    if (data.type === 'task:unassign') {
+      let { unassign_notifs, unassign_notifs_flush_timeout } = this.state,
+          new_timeout
+
+      if (unassign_notifs_flush_timeout) {
+        clearTimeout(unassign_notifs_flush_timeout)
+      }
 
-      Settings.addListener('server:remove', this.disconnect.bind(this));
+      unassign_notifs.push(data.task)
+      new_timeout = setTimeout(this.sendUnassignNotifs, 20*1000)
+      this.setState({ unassign_notifs: unassign_notifs.slice(), unassign_notifs_flush_timeout: new_timeout})
 
-      AppUser.load()
-        .then(user => {
-          Settings
-            .loadServer()
-            .then(baseURL => this.initializeRouter(baseURL, user))
-        })
+    } else if (data.type === 'task:assign') {
+      let { assign_notifs, assign_notifs_flush_timeout } = this.state,
+        new_timeout
+
+      if (assign_notifs_flush_timeout) {
+        clearTimeout(assign_notifs_flush_timeout)
+      }
+
+      assign_notifs.push(data.task)
+      new_timeout = setTimeout(this.sendAssignNotifs, 20*1000)
+      this.setState({ assign_notifs: assign_notifs.slice(), assign_notifs_flush_timeout: new_timeout})
     }
 
-    initializeRouter(baseURL, user) {
-      let client = null;
-      if (baseURL) {
-        client = API.createClient(baseURL, user);
+    // dispatch event if a specific screen has to deal with it
+    events.emit('websocket:message', event)
+  }
+
+  sendAssignNotifs () {
+    sendTasksAssignNotification(this.state.assign_notifs)
+    this.setState({ assign_notifs: [], assign_notifs_flush_timeout: null })
+  }
+
+  sendUnassignNotifs () {
+    sendTasksUnassignkNotification(this.state.unassign_notifs)
+    this.setState({ unassign_notifs: [], unassign_notifs_flush_timeout: null })
+  }
+
+  initializeRouter(baseURL, user) {
+    let client = null,
+      ws = null
+
+    if (baseURL) {
+      client = API.createClient(baseURL, user)
+      if (user && user.isAuthenticated() && (user.hasRole('ROLE_COURIER') || user.hasRole('ROLE_ADMIN'))) {
+        ws = this.createWebsocket(client)
```

Comment:
The WebSocket is created only if the user is already authenticated when launching the app. 
The following use case doesn't work: 
- Logout
- Close app
- Open app
- Login
- Go to tasks screen

In this case, `ws` is `undefined`, and if you go to the tasks screen, it hangs. 
We need to send an event when the user logs in, to create the WebSocket.